### PR TITLE
(PR 2) fix: create passcode only if using precall template

### PR DESCRIPTION
**Pull request by jia1**
## Problem

We should run passcode functionality and show passcode only if the message template is a pre-call template.

Closes [SGC-213](https://linear.app/ogp/issue/SGC-213/add-missed-call-and-house-visit-templates)

## Solution

- [x] Add if-condition before every creation of `GovsgVerification`
- [x] Change absent passcode text to `N/A`
- [x] Hide passcode column if the message template used is not a pre-call template
- [x] Temporarily disable Resend button for non-pre-call template messages

### Other improvements

- [x] Change `tr height` to `tr min-height` so that the message data column can expand vertically and we don't break other existing tables relying on `tr height`. Out-of-scope for this PR: Handling horizontal overflow

## Screenshots
Screenshot 2023-08-18 at 10 17 14 AM

### CODE REVIEW

File: backend/src/govsg/middlewares/govsg-verification.middleware.ts [hunk 1]
```diff
@@ -65,14 +65,23 @@ export const listMessages = async (
...
+    const row = rows[0]
+    return res.json({
+      messages,
+      total_count: count,
+      has_passcode: !!row.govsgVerification?.passcode,
+    })
+  }
   return res.json({
     messages,
     total_count: count,
```
Comment (KishenKumarrrrr) [inline backend/src/govsg/middlewares/govsg-verification.middleware.ts:83]: In this case, `messages` is an empty array and `count` is 0? Would it be clearer to explicitly state the values and add a comment explaining the case?

File: backend/src/govsg/services/govsg.service.ts [hunk 2]
```diff
@@ -139,6 +140,20 @@ const getLanguageCode = (language: string | undefined) => {
   return WhatsAppLanguages[key as keyof typeof WhatsAppLanguages]
 }
 
+const isUsingPrecallTemplate = async (govsgMessage: GovsgMessage) => {
```
Comment (KishenKumarrrrr) [inline backend/src/govsg/services/govsg.service.ts:143]: nit: Is this function name accurate? Having a passcode goes beyond just precall templates right? Perhaps `isTemplateWithPasscode` is better?

**Review by KishenKumarrrrr — COMMENTED** [review]
I have an additional question. If passcode is now optional, do we need to modify the `GovsgVerification` model?

**Comment by jia1** [comment]
> I have an additional question. If passcode is now optional, do we need to modify the `GovsgVerification` model?

Since some templates don't need passcode, then we don't even create `GovsgVerification` for those messages. This PR addresses that. Let me know if this doesn't answer your qn!

File: backend/src/govsg/services/govsg.service.ts (hunk 2 above)
Comment (jia1) [reply backend/src/govsg/services/govsg.service.ts:143]: For current use cases, the name is accurate. But sure, let me rename that.

File: backend/src/govsg/middlewares/govsg-verification.middleware.ts (hunk 1 above)
Comment (jia1) [reply backend/src/govsg/middlewares/govsg-verification.middleware.ts:83]: Yep. Got it, let me refactor this...

Comment (jia1) [reply backend/src/govsg/middlewares/govsg-verification.middleware.ts:75]: Done!

File: backend/src/govsg/services/govsg.service.ts (hunk 2 above)
Comment (jia1) [reply backend/src/govsg/services/govsg.service.ts]: Done!

File: backend/src/govsg/middlewares/govsg-verification.middleware.ts (hunk 1 above)
Comment (KishenKumarrrrr) [reply backend/src/govsg/middlewares/govsg-verification.middleware.ts:75]: Did you make the change? 🤔

Comment (jia1) [reply backend/src/govsg/middlewares/govsg-verification.middleware.ts:75]: I refactored and that block is not needed anymore. See 2nd latest commit.

Comment (KishenKumarrrrr) [reply backend/src/govsg/middlewares/govsg-verification.middleware.ts:75]: My bad, thanks for making the change.

**Pull request merged**